keps/1282-pods-ready-requeue-strategy/README.md

```diff
@@ -208,39 +209,48 @@ type RequeueState struct {
 }
 ```
 
-Add a new field `evictionStats` to the WorkloadStatus struct to track the number of evictions by type and reason:
+Add a new field, `schedulingStats`, to the WorkloadStatus struct to track the number of scheduling statistics.
+SchedulingState serves as a history of the Workload thus will not be reset.
+First sub field to be a part of the `schedulingStats` will be `eviction` a struct to track the number of evictions by reason.
 
 ```go
 type WorkloadStatus struct {
-	// evictionStats tracks eviction statistics by type and reason.
-    //
-    // +optional
-    // +listType=map
-    // +listMapKey=type
-    // +patchStrategy=merge
-    // +patchMergeKey=type
-    EvictionStats []Eviction `json:"evictionStats,omitempty"`
+	// schedulingStats tracks scheduling statistics (e.g. evictions)
+	//
+	// +optional
+	SchedulingStats *SchedulingStats `json:"schedulingStats,omitempty"`
 }
 
-type Eviction struct {
-    // type identifies the eviction type.
+type SchedulingStats struct {
+	// evictions tracks eviction statistics by reason and underlyingCause.
+	//
+	// +optional
+	// +listType=map
+	// +listMapKey=reason
+	// +listMapKey=underlyingCause
+	// +patchStrategy=merge
+	// +patchMergeKey=reason
+	// +patchMergeKey=underlyingCause
+	Evictions []WorkloadSchedulingStatsEvicition `json:"evictions,omitempty"`
+}
+
+type WorkloadSchedulingStatsEvicition struct {
```

Comment:
nit WorkloadSchedulingStatsEvicitions
to align with the field

Reply:
Couldn't we keep `WorkloadSchedulingStatsEvicition`? Because evicions are set of `WorkloadSchedulingStatsEvicition`.